### PR TITLE
feat: Support symmetric key token signature algorithm

### PR DESCRIPTION
**Pull request by JigarJoshi**
## Describe your changes
This PR adds support for symmetric key algorithm to verify token signature. This is required to decouple circular dependency amongst internal services.

Note: This PR changes the shape of config and needs attention with this release deployment.

## How best to test these changes
Added test case for this.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -36,8 +36,10 @@ RUN groupadd -r tigris && useradd -r -s /bin/false -g tigris tigris
 
 RUN mkdir -p /server /etc/tigrisdata/tigris /etc/foundationdb/
 
+ARG BUILD_PROFILE=""
```
Comment (rbarabas) [inline docker/Dockerfile:39]: Isn't there already an `env` based mechanism for this somewhere?

Comment (JigarJoshi) [reply docker/Dockerfile:39]: We need this at build time. Environment variables are available in container at runtime. For that reason we need build arg (i.e. build time env vars).

Let me know if I missed any existing mechanism to handle this.

File: server/middleware/auth.go [hunk 2]
```diff
@@ -97,14 +97,33 @@ func BypassAuthCaches(ctx context.Context) bool {
...
+	for i, issuerCfg := range config.Auth.Issuers {
+		log.Info().Msg(issuerCfg.Issuer)
+		issuerURL, _ := url.Parse(issuerCfg.Issuer)
+		var keyFunc func(ctx context.Context) (interface{}, error)
+		switch issuerCfg.Algorithm {
+		case validator.RS256:
+			{
+				provider := jwks.NewCachingProvider(issuerURL, config.Auth.JWKSCacheTimeout)
+				keyFunc = provider.KeyFunc
+				break
```
Comment (himank) [inline server/middleware/auth.go:109]: nit: You do not need braces and break in go inside cases.

File: server/middleware/auth.go [hunk 3]
```diff
@@ -97,14 +97,33 @@ func BypassAuthCaches(ctx context.Context) bool {
...
+		case validator.HS256:
+			{
+				keyFunc = func(ctx context.Context) (interface{}, error) {
+					return []byte(issuerCfg.SharedSecret), nil
+				}
+				break
+			}
+		default:
+			{
+				panic("Unsupported Token signature algorithm")
```
Comment (himank) [inline server/middleware/auth.go:120]: do we need to panic? Or return an error from here?

Comment (JigarJoshi) [reply server/middleware/auth.go:120]: This is the code path that will get run during startup and not per request - so panic here is fine.

File: server/middleware/auth.go (hunk 2 above)
Comment (JigarJoshi) [reply server/middleware/auth.go]: made update.

File: server/config/config.go [hunk 4]
```diff
@@ -86,6 +87,12 @@ func LoadConfig(config interface{}) {
 
 	// Deactivates lookup process for default server.yaml if
 	// configuration file is specified via the command line
+	if configFile != "" {
```
Comment (efirs) [inline server/config/config.go:90]:
```
if configFile == "" {
   configFile = os.Getenv(EnvTigrisConfigFile)
}
```

File: server/middleware/auth.go (hunk 3 above)
Comment (himank) [reply server/middleware/auth.go]: Okay, I thought it is in middleware and will be execute on per request.

**Review by himank — APPROVED** [review]

**Pull request merged**